From 4cc16829dbdf56cb1a77ea898f552429c308a1bf Mon Sep 17 00:00:00 2001
From: Tim Zimmermann <tim@linux4.de>
Date: Fri, 16 Sep 2022 08:27:44 +0200
Subject: [PATCH] universal2100-common: Pin R RIL blobs

* Signal strength does not get updated with S ones unless airplane mode
  is toggled on and off on T

Change-Id: Ie0d6d316457a0934b711497f082fbf8348a0e72b
---
 device_framework_matrix.xml |  2 +-
 extract-files.sh            |  3 +++
 manifest.xml                |  2 +-
 proprietary-files.txt       | 21 ++++++++++-----------
 vendor.prop                 |  1 +
 5 files changed, 16 insertions(+), 13 deletions(-)

diff --git a/device_framework_matrix.xml b/device_framework_matrix.xml
index 512edd0..ac61eb8 100644
--- a/device_framework_matrix.xml
+++ b/device_framework_matrix.xml
@@ -42,7 +42,7 @@
     </hal>
     <hal format="hidl">
         <name>vendor.samsung.hardware.radio</name>
-        <version>2.2</version>
+        <version>2.1</version>
         <interface>
             <name>ISehRadio</name>
             <instance>slot1</instance>
diff --git a/extract-files.sh b/extract-files.sh
index 099b378..7ff6058 100755
--- a/extract-files.sh
+++ b/extract-files.sh
@@ -66,6 +66,9 @@ function blob_fixup() {
         vendor/lib*/libexynosdisplay.so|vendor/lib*/hw/hwcomposer.exynos2100.so|vendor/lib*/sensors.*.so)
             "${PATCHELF}" --replace-needed libutils.so libutils-v32.so "${2}"
             ;;
+        vendor/bin/hw/rild)
+            "${PATCHELF}" --replace-needed libril.so libril-samsung.so "${2}"
+            ;;
     esac
 }
 
diff --git a/manifest.xml b/manifest.xml
index 2b1ca1c..8be5747 100644
--- a/manifest.xml
+++ b/manifest.xml
@@ -177,7 +177,7 @@
     <hal format="hidl">
         <name>vendor.samsung.hardware.radio</name>
         <transport>hwbinder</transport>
-        <version>2.2</version>
+        <version>2.1</version>
         <interface>
             <name>ISehRadio</name>
             <instance>slot1</instance>
diff --git a/proprietary-files.txt b/proprietary-files.txt
index c9b89a6..d705643 100644
--- a/proprietary-files.txt
+++ b/proprietary-files.txt
@@ -34,7 +34,6 @@ vendor/etc/usb_audio_policy_configuration.xml
 -vendor/lib/libprofileparamstorage.so
 -vendor/lib/libsecaudiocoreutils.so
 -vendor/lib/libsecaudioinfo.so
--vendor/lib/libsemnativecarrierfeature.so
 -vendor/lib/libspeakercalibration.so
 -vendor/lib/soundfx/libaudioeffectoffload.so
 -vendor/lib/soundfx/libaudiosaplus_sec.so
@@ -52,7 +51,6 @@ vendor/etc/usb_audio_policy_configuration.xml
 -vendor/lib64/libfloatingfeature.so
 -vendor/lib64/libsecaudiocoreutils.so
 -vendor/lib64/libsecaudioinfo.so
--vendor/lib64/libsemnativecarrierfeature.so
 -vendor/lib64/libspeakercalibration.so
 -vendor/lib64/soundfx/libaudioeffectoffload.so
 -vendor/lib64/soundfx/libaudiosaplus_sec.so
@@ -308,35 +306,36 @@ vendor/etc/init/android.hardware.neuralnetworks@1.3-service.eden-drv.rc
 
 # RIL
 -vendor/bin/cbd
--vendor/bin/hw/rild
 -vendor/lib/libaudio-ril.so
 -vendor/lib/libmdf.so
 -vendor/lib/libsec_semRil.so
 -vendor/lib/libsecnativefeature.so
 -vendor/lib/libsecril-client.so;MODULE=libsecril-client.exynos2100
 -vendor/lib/libsecure_storage.so
--vendor/lib64/libSemDataProps.so
--vendor/lib64/libSemTelephonyProps.so
 -vendor/lib64/libmdf.so
--vendor/lib64/libril_sem.so
 -vendor/lib64/libsec_semRil.so
 -vendor/lib64/libsecnativefeature.so
 -vendor/lib64/libsecril-client.so;MODULE=libsecril-client.exynos2100
 -vendor/lib64/libsecure_storage.so
 -vendor/lib64/vendor.samsung.hardware.radio.bridge@2.0.so
--vendor/lib64/vendor.samsung.hardware.radio.bridge@2.1.so
 -vendor/lib64/vendor.samsung.hardware.radio.channel@2.0.so
 -vendor/lib64/vendor.samsung.hardware.radio@2.0.so
 -vendor/lib64/vendor.samsung.hardware.radio@2.1.so
--vendor/lib64/vendor.samsung.hardware.radio@2.2.so
-vendor/etc/init/init.baseband.rc
-vendor/etc/init/init.vendor.rilchip.rc
-vendor/etc/init/init.vendor.rilcommon.rc
 vendor/etc/plmn_delta.bin
 vendor/etc/plmn_delta_attaio.bin
 vendor/etc/plmn_delta_usagsm.bin
 vendor/etc/plmn_se13.bin
 
+## RIL - from o1s - G991BXXU1AUA4
+vendor/etc/init/init.baseband.rc|8f2154bd1343d3ff90c465b087b449c6dfc4576b
+vendor/etc/init/init.vendor.rilchip.rc|83d57882dfaefaaa21c5dd636bafd655dfaf0b10
+vendor/etc/init/init.vendor.rilcommon.rc|eb3115e335457fdc95c854190ee6804e165718f2
+-vendor/bin/hw/rild|85f98906d824830d0e132fb63ea6adb1b99b08b9
+-vendor/lib/libril-samsung.so|24f397a3a38fab7828c46e74d17f3a3b0d7497ac
+-vendor/lib/libsemnativecarrierfeature.so|dcf01e550feb3a06ab82a47ee79641157dcddb8d
+-vendor/lib64/libril-samsung.so|3609d8c594525de92c97731d901b2fbc5633688f
+-vendor/lib64/libsemnativecarrierfeature.so|723634c41038c781b05550b32d623c37b91a7832
+
 # Surface Composer
 -vendor/lib/hw/hwcomposer.exynos2100.so
 -vendor/lib/libExynosHWCService.so
diff --git a/vendor.prop b/vendor.prop
index 22c7b02..92d3146 100644
--- a/vendor.prop
+++ b/vendor.prop
@@ -66,6 +66,7 @@ ro.vendor.nfc.support.ese=true
 
 # RIL
 vendor.sec.rild.libpath=/vendor/lib64/libsec-ril.so
+vendor.sec.rild.libpath2=/vendor/lib64/libsec-ril-dsds.so
 
 # Samsung specific
 ro.product_ship=true
